toolkit/block-producer-metadata/pallet/src/lib.rs

```diff
@@ -165,11 +181,12 @@ pub mod pallet {
 		#[pallet::call_index(0)]
 		#[pallet::weight(T::WeightInfo::upsert_metadata())]
 		pub fn upsert_metadata(
-			_origin: OriginFor<T>,
+			origin: OriginFor<T>,
 			metadata: T::BlockProducerMetadata,
 			signature: CrossChainSignature,
 			cross_chain_pub_key: CrossChainPublicKey,
 		) -> DispatchResult {
+			let origin_account = ensure_signed(origin)?;
```

Comment:
I think we had somewhere some verification of the maximum size of the data user can store, do you remember where it is?